### DISCUSSION TITLE
unable to get SX1262 working with raspberry pi pico and TTN.

### DISCUSSION THREAD

**Discussion by LucasWolf-Pudney** [jgromes/RadioLib#1305] [General]
Hi, I have the waveshare Pico-LoRa-SX1262-868M module and a raspberry pi pico. I am trying to get OTAA working with radiolib but I can not seem it get it working at all. I have also tried usng jumper wires to interface an ESP32 with the module and it still doenst work. I am using the LoRaWAN_starter example. and I am getting this error. 
Setup ... 
Initialise the radio
Join ('login') the LoRaWAN Network
Join failed - RADIOLIB_ERR_NO_JOIN_ACCEPT (-1116)

my end device is registered with EU 863-870. 1.1.0, regional params 1.1 A.
The only things I have changed in the config and .ino file are as follows:
changed 
SX1278 radio = new Module(10, 2, 9, 3);
to
SX1262 radio = new Module(3, 20, 15, 2, SPI1, RADIOLIB_DEFAULT_SPI_SETTINGS); // top of config.h
added this to the set up of the .ino

void setup() {
  Serial.begin(115200);

  
  while(!Serial);
  delay(5000);  // Give time to switch to the serial monitor
  Serial.println(F("\nSetup ... "));
  
  SPI1.setSCK(10);
  SPI1.setTX(11);
  SPI1.setRX(12);
  pinMode(3, OUTPUT);
  digitalWrite(3, HIGH);
  SPI1.begin(false);
  //---snip---

And changed the join EUI and others to my acivation infomation. 
These changes where all nessarary to get the module working and connecting as I was prevously getting chip not found errors. However, I can not see why the example should not work as it supports the SX1262 chipset along with the TTN paramters selected. 

Any help would be much appechated. 
thank you, 
Lucas

**Comment by jgromes** [comment]
I'm not a TTN or Raspberry Pi Pico expert, but I suspect even an expert will need some more information, mainly your code and build setup, whether you are using an Arduino or some Pico SDK ... etc.

**Comment by HeadBoffin** [comment]
> Does anyone know how to use an ESP32/pi pico with an SX1262 to do OTAA in radio lib.

Almost all the day to day testing of LW on RL is with an ESP32 & SX1262 combo - so it should work - but without that any logs it will be impossible to assist. Getting it working with an ESP32 first to check the hardware would be good as it will iron out / surface any LW / TTN issues as well.

**Comment by FourTweny** [comment]
Did you ever find out the solution?
